spec/lightstep_spec.rb

```diff
@@ -558,4 +558,87 @@ def init_callback_tracer(callback)
     expect(records[0][:span_name]).to eq("5")
     expect(records[1][:span_name]).to eq("[:foo]")
   end
+
+  describe '#scope_manager' do
+    let(:tracer) { init_test_tracer }
+
+    it 'should return a scope manager' do
+      expect(tracer.scope_manager).to be_an_instance_of(LightStep::ScopeManager)
+    end
+
+    context 'when the scope manager exists' do
+      before(:each) do
+        @manager = tracer.scope_manager
+      end
+
+      it 'should return the same scope manager' do
+        expect(tracer.scope_manager).to eq(@manager)
+      end
+    end
+  end
+
+  describe '#start_active_span' do
+    let(:tracer) { init_test_tracer }
+
+    it 'should set the active scope' do
+      span = tracer.start_active_span('some-operation')
+
+      actual = tracer.scope_manager.active
+      expect(actual).to be_an_instance_of(LightStep::Scope)
+      expect(actual.span.to_h[:span_name]).to eq('some-operation')
+    end
+
+    it 'should yield the active scope when given a block' do
+      tracer.start_active_span('some-operation') do |scope|
+        expect(scope).to be_an_instance_of(LightStep::Scope)
+
+        expected_scope = tracer.scope_manager.active
+        expect(scope).to eq(expected_scope)
+        expect(scope.span.to_h[:span_name]).to eq('some-operation')
+      end
+    end
+
+    context 'when there is an active scope' do
+      before(:each) do
+        @scope = tracer.start_active_span('some-operation')
+        @parent_span = @scope.span
+      end
+
+      it 'should create a child_of reference to the active scope' do
+        scope = tracer.start_active_span('child-operation')
+        expect(scope.span.tags[:parent_span_guid]).to eq(@parent_span.context.id)
+      end
+
+      context 'when ignore_active_scope is true' do
+        it 'should not create a child_of reference to the active scope' do
+          scope = tracer.start_active_span('child-operation', ignore_active_scope: true)
+          expect(scope.span.tags[:parent_span_guid]).not_to eq(@parent_span.context.id)
+        end
+      end
+    end
+
+    xit 'should create a span with the given operation_name'
+    xit 'should create a span with the given child_of'
+    xit 'should create a span with the given references'
+    xit 'should create a span with the given start_time'
+    xit 'should create a span with the given tags'
```

Comment:
These tests are pending because I'm not sure what a good way to test them would be. Tracer knows how to make a Span but asserting about the created span means digging into the internals of `LightStep::Span`, and that feels gross to me. Or, relying on the `LightStep::Span#to_h` method to make assertions, which also feels gross and reliant on just what `to_h` does. I'd rather not have to tie these tests to those implementations.

Reply:
I know it's not everyone's favourite testing style, but you can assert that the constructor was called correctly, e.g.:
```ruby
it 'should create a span with the given operation_name' do
  allow(LightStep::Span).to receive(:new).and_call_original # or return something else so that you can assert start_active_span actually returns the right thing, or just don't return anything, or...

  tracer.start_active_span('child-operation')

  expect(LightStep::Span).to have_received(:new).with(hash_including(operation_name: 'child-operation'))
end
```